### ISSUE TITLE
Document Adaptive Concurrency Controller

### ISSUE THREAD

**Issue by jszwedko** [timberio/vector#7971]
We have a nice blog post about this, https://vector.dev/blog/adaptive-request-concurrency/, but it'd be good to summarize and put the relevant details on  https://vector.dev/docs/about/under-the-hood/networking/adaptive-request-concurrency/ to help users understand how this feature works.

It should discuss:

* How this feature works
* How to monitor it

**Comment by zamazan4ik** [comment]
@jszwedko just a small fix: now the proper link is: https://vector.dev/docs/about/under-the-hood/networking/arc/

I have read the blog post and as a quick fix I can suggest just add a link to the article in the existing ARC documentation. Something like "For futher information and more deep details please read this one: https://vector.dev/blog/adaptive-request-concurrency/". What do you think?

Another way is copy more materials about under-the-hood work from the article into the documentaion.

Regarding the monitoring question. I think will be enough just mention that ARC monitoring should be done via `adaptive_concurrency_*` metrics from `internal_metrics` (https://vector.dev/docs/reference/configuration/sources/internal_metrics/).

What do you think?

**Comment by zamazan4ik** [comment]
By the way, according to the `adaptive_concurrency_*` metrics from `internal_metrics`, it describes global adaptive concurrency state across all sinks. Since each sink could have own adaptive concurrency state, the users could be interested in monitoring adaptive concurrency per sink, not globally.

**Comment by bruceg** [comment]
You are correct, the texts on the `adaptive_concurrency_*` metrics are misleading when read in the context of the `internal_metrics` source. All of those metrics are also tagged with the `component_*` tags to make them per-sink.

**Comment by zamazan4ik** [comment]
Thanks! I think we should document it somehow. Maybe somewhere on `internal_metrics` documentation page.

**Comment by bruceg** [comment]
I've opened #14351 to fix the missing bits.
